File: cpp/1221_SplitAStringInBalancedStrings.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int score = 0, count = 0;
        for (char c : s) {
            score += c == 'L' ? 1 : -1;
            if (score == 0) ++count;
        }
        return count;
    }
};

File: swift/1221_SplitAStringInBalancedStrings.swift
class Solution {
    func balancedStringSplit(_ s: String) -> Int {
        var score: Int = 0
        var count: Int = 0
        for c in s {
            score += c == "L" ? 1 : -1
            if score == 0 {
                count += 1
            }
        }
        return count
    }
}
